torch_geometric/datasets/proteinmpnn_dataset.py

```diff
@@ -0,0 +1,238 @@
+            }
+
+            if len(asmb_candidates) < 1:
+                chain = torch.load("%s_%s.pt" % (PREFIX, chid))
+                L = len(chain['seq'])
+                item_graph = Data(
+                    seq=chain['seq'],
+                    xyz=chain['xyz'],
+                    idx=torch.zeros(L).int(),
+                    masked=torch.Tensor([0]).int(),
+                    #label = self.item[0]
+                )
+
+                protein_graph.append(item_graph)
+
+            asmb_i = random.sample(list(asmb_candidates), 1)
+
+            idx = np.where(np.array(asmb_ids) == asmb_i)[0]
+
+            # load relevant chains
+            chains = {
+                c: torch.load("%s_%s.pt" % (PREFIX, c))
+                for i in idx
+                for c in asmb_chains[i] if c in meta['chains']
+            }
+
+            asmb = {}
+            for k in idx:
+
+                # pick k-th xform
+                xform = meta['asmb_xform%d' % k]
+                u = xform[:, :3, :3]
+                r = xform[:, :3, 3]
+
+                # select chains which k-th xform should be applied to
+                s1 = set(meta['chains'])
+                s2 = set(asmb_chains[k].split(','))
+                chains_k = s1 & s2
+
+                # transform selected chains
+                for c in chains_k:
+                    try:
+                        xyz = chains[c]['xyz']
+                        xyz_ru = torch.einsum('bij,raj->brai', u,
+                                              xyz) + r[:, None, None, :]
+                        asmb.update({
+                            (c, k, i): xyz_i
+                            for i, xyz_i in enumerate(xyz_ru)
+                        })
+                    except KeyError:
+                        item_graph = Data(seq=np.zeros(5))
+                        protein_graph.append(item_graph)
+                        continue
+
+            # select chains which share considerable similarity to chid
+            seqid = meta['tm'][chids == chid][0, :, 1]
+            homo = {
+                ch_j
+                for seqid_j, ch_j in zip(seqid, chids)
+                if seqid_j > params['HOMO']
+            }
+            # stack all chains in the assembly together
+            seq, xyz, idx, masked = "", [], [], []
+            seq_list = []
+            for counter, (k, v) in enumerate(asmb.items()):
+                seq += chains[k[0]]['seq']
+                seq_list.append(chains[k[0]]['seq'])
+                xyz.append(v)
+                idx.append(torch.full((v.shape[0], ), counter))
+                if k[0] in homo:
+                    masked.append(counter)
+
+            item_graph = Data(
+                seq=seq,
+                xyz=torch.cat(xyz, dim=0),
+                idx=torch.cat(idx, dim=0),
+                masked=torch.Tensor(masked).int(),
+                #label = self.item[0]
+            )
```

Comment:
I didn't see `x` and `edge_index` in the dataset, ProteinMPNN's input only includes sequence data, right?